mesonbuild/modules/gnome.py

```diff
@@ -406,12 +406,22 @@ def _get_gir_dep(self, state):
             kwargs = {'native': True, 'required': True}
             holder = self.interpreter.func_dependency(state.current_node, ['gobject-introspection-1.0'], kwargs)
             self.gir_dep = holder.held_object
-            if self.gir_dep.type_name == 'pkgconfig':
-                self.giscanner = ExternalProgram('g_ir_scanner', self.gir_dep.get_pkgconfig_variable('g_ir_scanner', {}))
-                self.gicompiler = ExternalProgram('g_ir_compiler', self.gir_dep.get_pkgconfig_variable('g_ir_compiler', {}))
+            giscanner = state.environment.lookup_binary_entry(MachineChoice.HOST, 'g-ir-scanner')
+            if giscanner is not None:
+                self.giscanner = ExternalProgram.from_bin_list(state.environment, MachineChoice.HOST, 'g-ir-scanner')
             else:
-                self.giscanner = self.interpreter.find_program_impl('g-ir-scanner')
-                self.gicompiler = self.interpreter.find_program_impl('g-ir-compiler')
+                if self.gir_dep.type_name == 'pkgconfig':
```

Comment:
nitpicking: we can save an indentation level with `elif`

Reply:
changed to use `elif`